packages/web-lib/utils/format.bigNumber.ts

```diff
@@ -59,21 +49,26 @@ export function	bigNumberAsAmount(
 	}${symbolWithPrefix}`);
 }
 
-export	const	toNormalizedValue = (v: BigNumberish, d?: number): number => (
-	Number(formatUnits(v || 0, d ?? 18))
-);
+export	const	toNormalizedValue = (v: `${number}` | string | bigint, d?: number): number => {
+	const vAsBigint = toBigInt(v);
+	return Number(formatUnits(vAsBigint, d ?? 18));
+};
 
-export const	toNormalizedAmount = (v: BigNumberish, d?: number): string => (
-	formatAmount(toNormalizedValue(v, d ?? 18), 6, 6)
-);
+export const	toNormalizedAmount = (v: `${number}`, d?: number): string => {
+	return formatAmount(toNormalizedValue(v, d ?? 18), 6, 6);
+};
 
-export const	toNormalizedBN = (value: BigNumberish, decimals?: number): TNormalizedBN => ({
-	raw: BN(value),
-	normalized: toNormalizedValue(BN(value), decimals ?? 18)
+export const	toNormalizedBN = (value: `${number}` | string | bigint, decimals?: number): TNormalizedBN => ({
+	raw: toBigInt(value),
+	normalized: toNormalizedValue(value, decimals ?? 18)
 });
 
+export function	parseUnits(value: `${number}` | string, decimals = 18): bigint {
+	const valueAsNumber = value as `${number}`;
```

Comment:
This is a bit of a strange cast, in the function signature we are saying that `value` can be a string, say `"foobar"`, but then when we enter the function we force it to be an interpolated number. So this would break.

I would remove the `string` from the `value` type for this to always work

Reply:
We need a lint rule to ban `as`, that's extremely dangerous and we're using it way too freely :)